mlflow/gluon.py

```diff
@@ -0,0 +1,214 @@
+    """
+    local_model_path = _download_artifact_from_uri(artifact_uri=model_uri)
+
+    model_arch_path = os.path.join(local_model_path, "data", _MODEL_SAVE_PATH) + "-symbol.json"
+    model_params_path = os.path.join(local_model_path, "data", _MODEL_SAVE_PATH) + "-0000.params"
+    symbol = sym.load(model_arch_path)
+    inputs = sym.var('data', dtype='float32')
+    net = gluon.SymbolBlock(symbol, inputs)
+    net.collect_params().load(model_params_path, ctx)
+    return net
+
+
+class _GluonModelWrapper:
+    def __init__(self, gluon_model):
+        self.gluon_model = gluon_model
+
+    def predict(self, ndarray):
+        return self.gluon_model(ndarray)
+
+
+def _load_pyfunc(path):
+    """
+    Load PyFunc implementation. Called by ``pyfunc.load_pyfunc``.
+
+    :param path: Local filesystem path to the MLflow Model with the ``gluon`` flavor.
+    """
+    m = load_model(path, mx.current_context())
+    return _GluonModelWrapper(m)
+
+
+@experimental
+def save_model(gluon_model, path, mlflow_model=Model()):
+    """
+    Save a Gluon model to a path on the local file system.
+
+    :param gluon_model: Gluon model to be saved. Must be already hybridized.
+    :param path: Local path where the model is to be saved.
+    :param mlflow_model: MLflow model config this flavor is being added to.
+
+    >>> from mxnet.gluon import Trainer
+    >>> from mxnet.gluon.contrib import estimator
+    >>> from mxnet.gluon.loss import SoftmaxCrossEntropyLoss
+    >>> from mxnet.gluon.nn import HybridSequential
+    >>> from mxnet.metric import Accuracy
+    >>> import mlflow
+    >>> # Build, compile, and train your model
+    >>> gluon_model_path = ...
+    >>> net = HybridSequential()
+    >>> with net.name_scope():
+    >>> ...
+    >>> net.hybridize()
+    >>> net.collect_params().initialize()
+    >>> softmax_loss = SoftmaxCrossEntropyLoss()
+    >>> trainer = Trainer(net.collect_params())
+    >>> est = estimator.Estimator(net=net, loss=softmax_loss, metrics=Accuracy(), trainer=trainer)
+    >>> est.fit(train_data=train_data, epochs=100, val_data=validation_data)
+    ... # Save the model as an MLflow Model
+    >>> mlflow.gluon.save_model(net, gluon_model_path)
+    """
+    path = os.path.abspath(path)
+    if os.path.exists(path):
+        raise MlflowException("Path '{}' already exists".format(path))
+    data_subpath = "data"
+    data_path = os.path.join(path, data_subpath)
+    os.makedirs(data_path)
+    # The epoch argument of the export method does not play any role in selecting
+    # a specific epoch's paramaters, and is there only for display purposes.
+    gluon_model.export(os.path.join(data_path, _MODEL_SAVE_PATH))
+    with open(os.path.join(path, "architecture.txt"), "w") as fp:
+        fp.write(str(gluon_model))
+    pyfunc.add_to_model(mlflow_model, loader_module="mlflow.gluon")
+    mlflow_model.save(os.path.join(path, "MLmodel"))
+
+
+@experimental
+def log_model(gluon_model, artifact_path):
```

Comment:
@cosmincatalin Can you add a `conda_env` parameter to the `save_model` method?

Reply:
Done